From 8b15d102e9f2b58679706a39702f8d18b9bdd00e Mon Sep 17 00:00:00 2001
From: Julian Andres Klode <jak@debian.org>
Date: Tue, 22 Jun 2010 12:53:47 +0200
Subject: [PATCH] Make it possible to run on sub directories only (Closes: #582114).

Introduce support for debian/autoreconf, a file which contains
a list of directories autoreconf should be run on. If present,
dh_autoreconf will only scan the given directories instead of
the complete source tree.
---
 dh_autoreconf |   22 +++++++++++++++++++++-
 1 files changed, 21 insertions(+), 1 deletions(-)

diff --git a/dh_autoreconf b/dh_autoreconf
index ad2e317..29c3bfa 100755
--- a/dh_autoreconf
+++ b/dh_autoreconf
@@ -21,6 +21,19 @@ debian/autoreconf.before and debian/autoreconf.after which contain checksums
 of all files before/after the build. It is complemented by dh_autoreconf_clean
 which creates a list of all changed and added files and removes them.
 
+=head1 FILES
+
+=over 4
+
+=item debian/autoreconf
+
+This file can contain a list of directories. If present, dh_autoreconf will
+only scan the given directories for changes. If no special command is given
+to dh_autoreconf, it will also forward the directory list to autoreconf
+which causes it to be run only on those sub directories.
+
+=back
+
 =head1 OPTIONS
 
 =over 4
@@ -64,11 +77,18 @@ shall not be run:
 
 init(options => { "mode=s" => \$dh{MODE}});
 
+my @directories;
 my $find_options='! -ipath "./debian/*" -a ';
 if (defined($dh{EXCLUDE_FIND}) && $dh{EXCLUDE_FIND} ne '') {
     $find_options .= "! \\( $dh{EXCLUDE_FIND} \\) -a";
 }
 
+if (-e "debian/autoreconf") {
+    @directories=filearray("debian/autoreconf", ".");
+    $find_options = join(" ",@directories)." ".$find_options;
+}
+
+
 my %modes = (
     'md5' => "find $find_options -type f -exec md5sum {} \\;",
     'timesize' => "find $find_options -type f -printf \"%s|%T@  %p\n\""
@@ -80,7 +100,7 @@ my $find = $modes{$dh{MODE} || "md5"} ||
 complex_doit("$find > debian/autoreconf.before");
 
 # Run autoreconf to recreate the needed files.
-@ARGV ? doit(@ARGV) : doit('autoreconf', '-f', '-i');
+@ARGV ? doit(@ARGV) : doit('autoreconf', '-f', '-i', @directories);
 
 complex_doit("$find > debian/autoreconf.after");
 
-- 
1.7.1

